File: Sources/MuseSaver/Support/Preferences.swift
import Foundation

/// App preferences persisted in UserDefaults.
enum Preferences {
    private static let showOnUnlockKey = "showOnUnlock"

    /// Whether the lock screen should appear automatically when the Mac unlocks.
    /// Defaults to true.
    static var showOnUnlock: Bool {
        get {
            if UserDefaults.standard.object(forKey: showOnUnlockKey) == nil { return true }
            return UserDefaults.standard.bool(forKey: showOnUnlockKey)
        }
        set { UserDefaults.standard.set(newValue, forKey: showOnUnlockKey) }
    }

    private static let syncLEDsKey = "syncLEDs"

    /// Whether to sync BLE LED strips to the album color. Defaults to false.
    static var syncLEDs: Bool {
        get { UserDefaults.standard.bool(forKey: syncLEDsKey) }
        set { UserDefaults.standard.set(newValue, forKey: syncLEDsKey) }
    }
}

File: Windows/Support/Preferences.cs
using System.IO;
using System.Text.Json;

namespace MuseSaverWin.Support;

/// <summary>App preferences persisted as JSON under %LOCALAPPDATA%\MuseSaver.</summary>
internal static class Preferences
{
    private sealed class Data
    {
        public bool ShowOnUnlock { get; set; } = true;
    }

    private static readonly string FilePath = Path.Combine(
        Environment.GetFolderPath(Environment.SpecialFolder.LocalApplicationData),
        "MuseSaver", "prefs.json");

    private static Data _data = Load();

    private static Data Load()
    {
        try
        {
            if (File.Exists(FilePath))
            {
                var json = File.ReadAllText(FilePath);
                var loaded = JsonSerializer.Deserialize<Data>(json);
                if (loaded != null) return loaded;
            }
        }
        catch
        {
            // Corrupt or unreadable prefs file — fall back to defaults.
        }
        return new Data();
    }

    private static void Save()
    {
        try
        {
            var dir = Path.GetDirectoryName(FilePath)!;
            Directory.CreateDirectory(dir);
            File.WriteAllText(FilePath, JsonSerializer.Serialize(_data));
        }
        catch
        {
            // Best-effort persistence; nothing user-actionable to do on failure.
        }
    }

    /// <summary>Whether the lock screen should appear automatically when Windows unlocks.</summary>
    public static bool ShowOnUnlock
    {
        get => _data.ShowOnUnlock;
        set
        {
            _data.ShowOnUnlock = value;
            Save();
        }
    }
}
